dash-renderer/src/APIController.react.js

```diff
@@ -23,15 +28,21 @@ import {STATUS} from './constants/constants';
 const UnconnectedContainer = props => {
     const [errorLoading, setErrorLoading] = useState(false);
 
+    const events = useRef(null);
+    if (!events.current) {
+        events.current = new EventEmitter();
+    }
+    const renderedTree = useRef(false);
+
     useEffect(() => {
         const {
             appLifecycle,
             dependenciesRequest,
             dispatch,
+            error,
```

Comment:
This component might be easier to read if we make this effect a standalone function
```
function storeEffect(props) { ... }

useEffect(storeEffect.call(this, props))
```

Reply:
sure -> https://github.com/plotly/dash/pull/1103/commits/883a6ef80ad42f0ba0ec640a95deb66189bf031f